### PR TITLE
added ts-loader to webpack config

### PR DESCRIPTION
**Pull request by adufilie**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -110,8 +117,17 @@ var config = {
...
+        preLoaders: [
+            // All output '.js' files will have any sourcemaps re-processed by 'source-map-loader'.
+            {
+                test: /\.js$/,
+                loader: "source-map-loader"
+            }
+        ]
     },
     'postcss': [require('autoprefixer')],
+    'devtool': 'source-map',
```
Comment (alisman) [inline webpack.config.js]: in latest mast there's an optimization around sourcemap generation.
when we merge, lets make sure we retain it.

Comment (adufilie) [inline webpack.config.js]: I followed the instructions [here](http://www.typescriptlang.org/docs/handbook/react-&-webpack.html).  Is this line not required?

Comment (adufilie) [inline webpack.config.js]: What is the optimization?

**Comment by adufilie** [comment]
removed the uses of es7 object rest spread because it's not supported by TypeScript yet and it causes errors to appear in IntelliJ

File: .eslintrc.json [hunk 2]
```diff
@@ -8,7 +8,6 @@
 
     "parserOptions": {
         "ecmaFeatures": {
-            "experimentalObjectRestSpread": true,
```
Comment (inodb) [inline .eslintrc.json]: Would it be possible to keep the spread syntax? I've used it a few times. Ok with tossing it if necessary

File: dist/index.html [hunk 3]
```diff
@@ -5,10 +5,10 @@
...
     <script>window.defaultRoute = "/patient";</script>
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.2.4/jquery.min.js"></script>
-<link href="/reactapp/css/styles.css" rel="stylesheet"></head>
+</head>
 <body class="cbioOnePageApp">
     <span id="reactHeader"></span>
     <div id="clinical_div"></div>
     <div id="reactRoot"></div>
-<script type="text/javascript" src="/reactapp/js/main.app.js"></script></body>
+<script type="text/javascript" src="http://localhost:3000/reactapp/js/main.app.js"></script></body>
```
Comment (inodb) [inline dist/index.html]: why is this necessary? Windows thing? I think we should try not to add ports in the source file, since you should be able to override it with env variables on build. Don't feel particularly strongly about it tho, since this is just a dev file.

File: webpack.config.js (hunk 1 above)
Comment (alisman) [inline webpack.config.js]: The devtool property is actually being overwritten down below on env specific basis.  As long as it's in dev, i'm fine with any sourcemap option that is relatively performant (on build).  i had chosen inline-sourcemap.  But not with any great reason.

File: dist/index.html (hunk 3 above)
Comment (adufilie) [inline dist/index.html]: I didn't do this on purpose.. Not sure what happened

File: .eslintrc.json (hunk 2 above)
Comment (adufilie) [inline .eslintrc.json]: Typescript does not allow it. It created ts errors in intellij

File: dist/index.html (hunk 3 above)
Comment (adufilie) [inline dist/index.html]: This is in the dist folder btw

Comment (inodb) [inline dist/index.html]: 👌! Actually Travis should've caught this if it was only specific to your build env. Maybe something in build process that changed?

File: .eslintrc.json (hunk 2 above)
Comment (inodb) [inline .eslintrc.json]: Like nothing in TS works anymore, or it just gives some warning on that line in intellij?

Comment (adufilie) [inline .eslintrc.json]: TS is unable to parse the file
